### ISSUE TITLE
Adaptive line search for Frank-Wolfe breaks tests

### ISSUE THREAD

**Issue by gdalle** [JuliaDecisionFocusedLearning/InferOpt.jl#40]
Frank-Wolfe related tests fail due to the linear solver for implicit differentiation not converging

Main suspect: https://github.com/ZIB-IOL/FrankWolfe.jl/issues/387

**Comment by gdalle** [comment]
Possible solution: #39

**Comment by gdalle** [comment]
We still need to diagnose why this happened in the first place

**Comment by gdalle** [comment]
Diagnosis will be easier once #65 is merged. If we switch the line search strategy in the new tests I think the bugs pop up again

**Comment by BatyLeo** [comment]
@gdalle the latest `FrankWolfe.jl` release seems to have broken our `RegularizedFrankWolfe` tests.
I'm facing the following error locally on main (on paths and ranking tests, argmax seems fine):
```julia
 Warning: IterativeLinearSolver failed, result contains NaNs
└ @ ImplicitDifferentiation ~/.julia/packages/ImplicitDifferentiation/HJshL/src/linear_solver.jl:34
```

**Comment by gdalle** [comment]
It may also be due to the latest release of ImplicitDifferentiation. I will take a look

**Comment by BatyLeo** [comment]
Fixed by #116

**Comment by gdalle** [comment]
well done!
